src/Microsoft.Unity.Analyzers/SetPositionAndRotation.cs

```diff
@@ -0,0 +1,241 @@
+
+			var leftSymbol = model.GetSymbolInfo(left);
+
+			if (leftSymbol.Symbol == null)
+				return false;
+
+			if (!(leftSymbol.Symbol is IPropertySymbol))
+				return false;
+
+			var leftExpressionTypeInfo = model.GetTypeInfo(left.Expression);
+
+			if (leftExpressionTypeInfo.Type == null)
+				return false;
+
+			if (!leftExpressionTypeInfo.Type.Extends(typeof(UnityEngine.Transform)))
+				return false;
+
+			return true;
+
+		}
+	}
+
+	[ExportCodeFixProvider(LanguageNames.CSharp)]
+	public class SetPositionAndRotationCodeFix : CodeFixProvider
+	{
+		public sealed override ImmutableArray<string> FixableDiagnosticIds => ImmutableArray.Create(SetPositionAndRotationAnalyzer.Rule.Id);
+
+		public sealed override FixAllProvider GetFixAllProvider() => WellKnownFixAllProviders.BatchFixer;
+
+		public sealed override async Task RegisterCodeFixesAsync(CodeFixContext context)
+		{
+			var root = await context.Document.GetSyntaxRootAsync(context.CancellationToken).ConfigureAwait(false);
+
+			if (!(root?.FindNode(context.Span) is AssignmentExpressionSyntax expression))
+				return;
+
+			context.RegisterCodeFix(
+				CodeAction.Create(
+					Strings.SetPositionAndRotationCodeFixTitle,
+					ct => ReplaceWithInvocationAsync(context.Document, expression, "transform", "SetPositionAndRotation", ct),
+					expression.ToFullString()),
+				context.Diagnostics);
+		}
+		private static async Task<Document> ReplaceWithInvocationAsync(Document document, AssignmentExpressionSyntax assignmentExpression, string identifierName, string genericMethodName, CancellationToken cancellationToken)
+		{
+			var root = await document.GetSyntaxRootAsync(cancellationToken).ConfigureAwait(false);
+
+			if (assignmentExpression.FirstAncestorOrSelf<BlockSyntax>() == null)
+				return document;
+
+			var block = assignmentExpression.FirstAncestorOrSelf<BlockSyntax>();
+
+			if (assignmentExpression.FirstAncestorOrSelf<ExpressionStatementSyntax>() == null)
+				return document;
+
+			var expression = assignmentExpression.FirstAncestorOrSelf<ExpressionStatementSyntax>();
+
+			var siblingsAndSelf = block.ChildNodes().ToImmutableArray();
+
+			if (siblingsAndSelf.LastIndexOf(expression) == -1)
+				return document;
+
+			var currentIndex = siblingsAndSelf.LastIndexOf(expression);
+
+			var nextIndex = currentIndex + 1;
+
+			if (nextIndex == siblingsAndSelf.Length)
+				return document;
+
+			var statement = siblingsAndSelf[nextIndex];
+
+			if (!(statement is ExpressionStatementSyntax))
+				return document;
+
+			var nextExpression = ((ExpressionStatementSyntax)statement).Expression;
+
+
+			if (!(nextExpression is AssignmentExpressionSyntax))
+				return null;
+
+			var nextAssignmentExpression = (AssignmentExpressionSyntax)nextExpression;
+
+			if (assignmentExpression.Right == null || nextAssignmentExpression.Right == null)
+				return document;
+
+			var argList = ArgumentList();
+
+			var property = ((MemberAccessExpressionSyntax)(assignmentExpression.Left)).Name.ToString();
```

Comment:
Redundant parenthesis :

```suggestion
			var property = ((MemberAccessExpressionSyntax)assignmentExpression.Left).Name.ToString();
```